client/transaction-pool/src/enactment_state.rs

```diff
@@ -18,13 +18,17 @@
 
 //! Substrate transaction pool implementation.
 
+use num_traits::CheckedSub;
 use sc_transaction_pool_api::ChainEvent;
 use sp_blockchain::TreeRoute;
-use sp_runtime::traits::Block as BlockT;
+use sp_runtime::traits::{Block as BlockT, NumberFor};
+
+const SKIP_MAINTENANCE_THRESHOLD: u16 = 20;
```

Comment:
```suggestion
/// The threshold since the last update where we will skip any maintenance for blocks.
/// 
/// This includes tracking re-orgs and sending out certain notifications. In general this shouldn't happen
/// and may only happen when the node is doing a full sync.
const SKIP_MAINTENANCE_THRESHOLD: u16 = 20;
```